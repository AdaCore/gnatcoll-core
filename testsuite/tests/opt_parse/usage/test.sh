STDOUT=stdout.log

./test > $STDOUT

# Check that the help text contains updated usage information
if grep -q "[--charset|-C <charset name>]" $STDOUT; then
   echo "stdout contained new usage text for option"
else
   echo "stdout missing new usage text for option"
fi

if grep -q "[--day|-D <three letter day of week>]" $STDOUT; then
   echo "stdout contained new usage text for enum option"
else
   echo "stdout missing new usage text for enum option"
fi

if grep -q "[--files|-F <list of filepaths to parse>]" $STDOUT; then
   echo "stdout contained new usage text for option list"
else
   echo "stdout missing new usage text for option list"
fi

# Check that the help text contains default usage information
if grep -q "[--charset2|-C2 CHARSET2]" $STDOUT; then
   echo "stdout contained default usage text for option"
else
   echo "stdout missing default usage text for option"
fi

if grep -q "[--day2|-D2 DAY2]" $STDOUT; then
   echo "stdout contained default usage text for enum option"
else
   echo "stdout missing default usage text for enum option"
fi

if grep -q "[--files2|-F2 FILES2[FILES2...]]" $STDOUT; then
   echo "stdout contained default usage text for option list"
else
   echo "stdout missing default usage text for option list"
fi
