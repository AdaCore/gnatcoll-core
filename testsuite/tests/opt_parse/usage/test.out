stdout contained new usage text for option
stdout contained new usage text for enum option
stdout contained new usage text for option list
stdout contained default usage text for option
stdout contained default usage text for enum option
stdout contained default usage text for option list